internal/expr/expr_test.go

```diff
@@ -484,24 +503,24 @@ func (s *ExprSuite) TestPrepareErrors(c *C) {
 		err:         `cannot prepare statement: output expression: type "Address" not passed as a parameter (have "Person"): street AS &Address.id`,
 	}, {
 		query:       "SELECT * AS &Person.* FROM t",
-		prepareArgs: []any{[]any{Person{}}},
-		err:         `cannot prepare statement: need struct or map, got slice`,
+		prepareArgs: []any{func() {}},
+		err:         `cannot prepare statement: need struct, map or slice, got func`,
 	}, {
 		query:       "SELECT * AS &Person.* FROM t",
 		prepareArgs: []any{&Person{}},
-		err:         `cannot prepare statement: need struct or map, got pointer to struct`,
+		err:         `cannot prepare statement: need struct, map or slice, got pointer to struct`,
 	}, {
 		query:       "SELECT * AS &Person.* FROM t",
 		prepareArgs: []any{(*Person)(nil)},
-		err:         `cannot prepare statement: need struct or map, got pointer to struct`,
+		err:         `cannot prepare statement: need struct, map or slice, got pointer to struct`,
 	}, {
 		query:       "SELECT * AS &Person.* FROM t",
 		prepareArgs: []any{map[string]any{}},
 		err:         `cannot prepare statement: cannot use anonymous map`,
 	}, {
 		query:       "SELECT * AS &Person.* FROM t",
 		prepareArgs: []any{nil},
-		err:         `cannot prepare statement: need struct or map, got nil`,
+		err:         `cannot prepare statement: need struct, map or slice, got nil`,
```

Comment:
We're getting close to it being "need [every possible type] but got nil". :)  How about "need valid value, got nil" for these cases instead?

Reply:
Changed the `nil` case to "need valid value, got nil". 

I've also changed the other errors where the user provides the wrong type (e.g. int) from:
"need struct, map or slice, got int"
to 
"unsupported type: int"